# HG changeset patch
# User Vytautas Janƒçauskas <unaudio@gmail.com>
# Date 1378916090 -10800
#      Wed Sep 11 19:14:50 2013 +0300
# Node ID e30c88336ee9052da972cf806e19114d18b71b77
# Parent  3c35ee570366724667a78d360b980fe886eedbf3
Added checks for sndfile in configure.ac

* configure.ac: added checks for libsndfile, and export HAVE_SNDFILE if
sndfile is found

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -931,22 +931,37 @@ if test -z "$warn_curl"; then
       [Define to the legacy option name if using an older version of cURL.])
   fi
 fi
 LIBS="$save_LIBS"
 CPPFLAGS="$save_CPPFLAGS"
 
 ### Check for sndfile
 
-SNDFILE_CPPFLAGS=`$PKG_CONFIG --cflags-only-I sndfile`
-SNDFILE_LDFLAGS=`$PKG_CONFIG --libs-only-L sndfile`
-SNDFILE_LIBS=`$PKG_CONFIG --libs-only-l sndfile`
-
-CPPFLAGS="$SNDFILE_CPPFLAGS $CPPFLAGS"
-LIBS="$SNDFILE_LDFLAGS $SNDFILE_LIBS $LIBS"
+SNDFILE_CPPFLAGS=
+SNDFILE_LDFLAGS=
+SNDFILE_LIBS=
+
+warn_sndfile="$sndfile library fails tests.  The audioinfo, audioread and audiowrite functions for reading and writing audio files will not be fully functional."
+
+PKG_CHECK_EXISTS([sndfile], [
+  SNDFILE_CPPFLAGS=`$PKG_CONFIG --cflags-only-I sndfile`
+  SNDFILE_LDFLAGS=`$PKG_CONFIG --libs-only-L sndfile`
+  SNDFILE_LIBS=`$PKG_CONFIG --libs-only-l sndfile`
+  warn_sndfile=
+])
+
+if test -z "$warn_sndfile"; then
+  AC_DEFINE(HAVE_SNDFILE, 1,
+    [Define to 1 if sndfile is available.])
+else
+  SNDFILE_CPPFLAGS=
+  SNDFILE_LDFLAGS=
+  SNDFILE_LIBS=
+fi
 
 AC_SUBST(SNDFILE_CPPFLAGS)
 AC_SUBST(SNDFILE_LDFLAGS)
 AC_SUBST(SNDFILE_LIBS)
 
 ### Check for either of Graphics/ImageMagick++ libraries
 
 AC_ARG_WITH([magick],
